src/api/cvc4cpp.cpp

```diff
@@ -5678,6 +5678,74 @@ SmtEngine* Solver::getSmtEngine(void) const { return d_smtEngine.get(); }
  */
 Options& Solver::getOptions(void) { return d_smtEngine->getOptions(); }
 
+/* -------------------------------------------------------------------------- */
+/* Optimization                                                               */
+/* -------------------------------------------------------------------------- */
+
+Objective:: Objective(Term t, ObjectiveType type){
+  d_type = type;
+  d_term = t;
+  d_result = OPT_UNKNOWN;
+}
+
+/**
+ * makes a minimize objective
```

Comment:
(please verify by looking at other functions, but) I think we typically document functions in the h files and not cpp. Comments in the cpp tend to be regarding the implementation.